src/iomusxml.cpp

```diff
@@ -1003,6 +1005,36 @@ bool MusicXmlInput::ReadMusicXmlMeasure(
         }
     }
 
+    // match open ties with close ties
+    double lastScoreTimeOnset;
+    bool tieMatched;
+    std::vector<std::pair<Tie *, Note *> >::iterator iter;
+    for (iter = m_tieStack.begin(); iter != m_tieStack.end(); ++iter) {
+        lastScoreTimeOnset = 9999; // __DBL_MAX__;
+        tieMatched = false;
+        std::vector<Note *>::iterator jter;
+        for (jter = m_tieStopStack.begin(); jter != m_tieStopStack.end(); ++jter) {
+            // match tie stop with pitch/oct identity, with start note earlier than end note,
+            // and with earliest end note.
+            if (iter->second->GetPname() == (*jter)->GetPname() && iter->second->GetOct() == (*jter)->GetOct()
+                && (iter->second->GetScoreTimeOnset() < (*jter)->GetScoreTimeOnset()
```

Comment:
Just to make sure I understand: Note::m_scoreTimeOnset is a MIDI value. I can see that you use it here. We need to make sure it does not have a side effect in MIDI output. (probably it does not.)

Reply:
I have added it to a note here: https://github.com/wergo/verovio/blob/edc9e2b4c1d8a09956c9d59ab080deb58d55659c/src/iomusxml.cpp#L1612

It has not been stored before, so I doubt that it will be read later. Where should I check?

Reply:
To clarify: I have called it MIDI value, because I read it somewhere in the code. It is actually the tick time that musicXML defines.

Reply:
Is there an easy way to reset the value to 0.0 once this is done? Then we would be sure there is not side effect.
Alternatively, we can reset it before we run MIDI export. I can look at it.

Reply:
The reason, I set it to `-1` is that in the next measure it is smaller than any of those notes coming. If there is a measure offset (start beat number of each measure or just a measure count that we could mutliply with the time signature), then I could just use that to order notes by their onset across measures.

Reply:
If we add an `int i` to `measure.h` that is just counting up the bar number from the beginning (e.g., in https://github.com/rism-ch/verovio/blob/f46d2fabd815d098488d6c713f38f93335f4cef5/src/iomusxml.cpp#L920) that we could easily reset `scoreTimeOnset=0.0` of a note and still have an ordering across measures.

Reply:
I think we can easily reset it in LayerElement::CalcOnsetOffset. This is all good